src/silx/io/dictdumplinks/_schemas.py

```diff
@@ -0,0 +1,24 @@
+from typing import cast
+from collections.abc import Mapping
+
+from ._vds_types import VdsSchemaV1
+from ._vds_types import VDSLink
+from ._vds_types import deserialize_vds_schema_v1
+
+from ._external_binary_types import ExtSchemaV1
+from ._external_binary_types import ExternalBinaryLink
+from ._external_binary_types import deserialize_ext_schema_v1
+
+
+def deserialize_schema(target: Mapping) -> VDSLink | ExternalBinaryLink | None:
+    """Convert a mapping into a link when it describes a link.
+    Otherwise return `None`.
+    """
+    dictdump_schema = target.get("dictdump_schema")
+    if dictdump_schema == "external_binary_link_v1":
+        ext_model = deserialize_ext_schema_v1(cast(ExtSchemaV1, target))
+        return ExternalBinaryLink(ext_model.shape, ext_model.dtype, ext_model.sources)
+    if dictdump_schema == "virtual_dataset_v1":
+        vds_model = deserialize_vds_schema_v1(cast(VdsSchemaV1, target))
+        return VDSLink(vds_model.shape, vds_model.dtype, vds_model.sources)
+    return None
```

Comment:
Distinguishing any dict from a schema is currently done by the presence of a field called `dictdump_schema`. If it has specific values it is a schema, otherwise it is an HDF5 group.

In addition this only gets called for values from keys which start with `>`.